src/pages/FlagCommentPage.tsx

```diff
@@ -51,6 +53,16 @@ function FlagCommentPage({parentReportAction, route, report, parentReport, repor
     const styles = useThemeStyles();
     const {translate} = useLocalize();
     const isReportArchived = useReportIsArchived(report?.reportID);
+    const [allReports] = useOnyx(ONYXKEYS.COLLECTION.REPORT, {canBeMissing: true});
+    let reportID: string | undefined = getReportID(route);
+
+    // Handle threads if needed
+    if (isChatThread(report) && reportAction?.reportActionID === parentReportAction?.reportActionID) {
+        reportID = parentReport?.reportID;
+    }
+    const originalReportID = getOriginalReportID(reportID, reportAction);
+    const originalReport = allReports?.[`${ONYXKEYS.COLLECTION.REPORT}${originalReportID}`];
```

Comment:
Here is another case where I don't think it should be connecting to the entire collection.